syncd/FlexCounter.cpp

```diff
@@ -1367,6 +1417,54 @@ void FlexCounter::collectPriorityGroupAttrs(
     }
 }
 
+void FlexCounter::collectMACsecSAAttrs(
+        _In_ swss::Table &countersTable)
+{
+    SWSS_LOG_ENTER();
+
+    // Collect attrs for every registered MACsec SA
+    for (const auto &kv: m_macsecSAAttrIdsMap)
+    {
+        const auto &macsecSAVid = kv.first;
+        const auto &macsecSARid = kv.second->m_macsecSAId;
+        const auto &macsecSAAttrIds = kv.second->m_macsecSAAttrIds;
+
+        std::vector<sai_attribute_t> macsecSAAttr(macsecSAAttrIds.size());
+
+        for (size_t i = 0; i < macsecSAAttrIds.size(); i++)
+        {
+            macsecSAAttr[i].id = macsecSAAttrIds[i];
+        }
+
+        // Get MACsec SA attr
+        sai_status_t status = m_vendorSai->get(
+                SAI_OBJECT_TYPE_MACSEC_SA,
+                macsecSARid,
+                static_cast<uint32_t>(macsecSAAttr.size()),
+                macsecSAAttr.data());
+
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            SWSS_LOG_WARN("Failed to get attr of MACsec SA 0x%" PRIx64 ": %d", macsecSAVid, status);
```

Comment:
use sai_serialize_object_id() and sai_serialize_status()

Reply:
Done, please review it.